src/NRedisStack/Auxiliary.cs

```diff
@@ -26,20 +30,62 @@ public static object[] AssembleNonNullArguments(params object?[] arguments)
             return args.ToArray();
         }
 
+        // public static IDatabase GetDatabase(this ConnectionMultiplexer redis) => redis.GetDatabase("", "");
+
+        // TODO: add all the signatures of GetDatabase
+        public static IDatabase GetDatabase(this ConnectionMultiplexer redis,
+                                            string? LibraryName)
+        {
+            var _db = redis.GetDatabase();
+            if (LibraryName == null) // the user wants to disable the library name and version sending
+                _libraryName = null;
+
+            else // the user set his own the library name
+                _libraryName = $"NRedisStack({LibraryName});.NET-{Environment.Version})";
+
+            return _db;
+        }
+
+        // TODO: understand why this method is not working:
+        // private static void SetInfoInPipeline(this IDatabase db)
+        // {
+        //     if (_libraryName == null) return;
+        //     Pipeline pipeline = new Pipeline(db);
+        //     _ = pipeline.Db.ClientSetInfoAsync(SetInfoAttr.LibraryName, _libraryName!);
+        //     _ = pipeline.Db.ClientSetInfoAsync(SetInfoAttr.LibraryVersion, GetNRedisStackVersion());
+        //     pipeline.Execute();
+        // }
+
         public static RedisResult Execute(this IDatabase db, SerializedCommand command)
         {
+            var compareVersions = db.Multiplexer.GetServer(db.Multiplexer.GetEndPoints()[0]).Version.CompareTo(new Version(7, 1, 242));
+            if (_setInfo && compareVersions >= 0)
+            {
+                _setInfo = false;
```

Comment:
also, your tests don't work in conjunction because `_setInfo` is static and never being reverted :)